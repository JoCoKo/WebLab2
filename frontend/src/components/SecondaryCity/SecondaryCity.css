.SecondaryCity {
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    width: 490px;
    height: 310px;
}

.SecondaryInfo {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
    height: 80px;
}

.SecondaryInfo span {
    font-size: 31px;
    font-weight: bold;
}

.SecondaryInfo div {
    display: flex;
    align-items: center;
}

#temp{
    color: rgb(124, 124, 124);
}