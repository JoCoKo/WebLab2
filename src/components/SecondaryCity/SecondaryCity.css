.SecondaryCity {
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    min-width: 45%;
}

.SecondaryInfo {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-around;
}

.SecondaryInfo span {
    font-size: 31px;
    font-weight: bold;
}

.SecondaryInfo div {
    display: flex;
    align-items: center;
}

.SecondaryCity button {
    width: 30px;
}