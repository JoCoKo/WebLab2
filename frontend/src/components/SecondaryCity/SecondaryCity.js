import React from 'react';
import './SecondaryCity.css';
import List from "../List/List";
import Spinner from "../Spinner/Spinner";


const SecondaryCity = ({weather, onDeleted, i}) => {
    const {city, temp, image, isLoading} = weather;
    return (
        <div className='SecondaryCity'>
            { isLoading ?
                <Spinner/> :
                <>
                <div className='SecondaryInfo'>
                    <span>{city}</span>
                    <img src={image} alt=''/>
                    <span id='temp'>{temp} â„ƒ</span>
                    <button onClick={() => onDeleted(i)}>X</button>
                </div>
                <List {...weather}/>
                </>
            }
        </div>
    );
};

export default SecondaryCity;
