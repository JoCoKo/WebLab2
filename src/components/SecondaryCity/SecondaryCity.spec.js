import React from 'react';
import SecondaryCity from "./SecondaryCity"
import {shallowToJson} from "enzyme-to-json";
import {shallow} from "enzyme";

describe('SecondaryCity', () => {
  describe('в состоянии загрузки', () => {
    const props = {
      weather: {
        city: "Ufa",
        temp: "-6",
        image: "http://openweathermap.org/img/wn/04n@2x.png",
        isLoading: true
      },
      onDeleted: ()=>{},
      i: 1
    };

    const secondaryCity = shallow(<SecondaryCity {...props} />);
    test('отображается корректно', () => {
      expect(shallowToJson(secondaryCity)).toMatchSnapshot();
    });
  });

  describe('в состоянии штатного отображения', () => {
    const props = {
      weather: {
        city: "Ufa",
        temp: "-6",
        image: "http://openweathermap.org/img/wn/04n@2x.png",
        isLoading: null
      },
      onDeleted: ()=>{},
      i: 1
    };

    const secondaryCity = shallow(<SecondaryCity {...props} />);
    test('отображается корректно', () => {
      expect(shallowToJson(secondaryCity)).toMatchSnapshot();
    });
  });
});