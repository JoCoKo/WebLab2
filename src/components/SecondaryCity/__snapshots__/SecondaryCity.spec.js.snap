// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondaryCity в состоянии загрузки отображается корректно 1`] = `
<div
  className="SecondaryCity"
>
  <Spinner />
</div>
`;

exports[`SecondaryCity в состоянии штатного отображения отображается корректно 1`] = `
<div
  className="SecondaryCity"
>
  <div
    className="SecondaryInfo"
  >
    <span>
      Ufa
    </span>
    <img
      alt=""
      src="http://openweathermap.org/img/wn/04n@2x.png"
    />
    <span
      id="temp"
    >
      -6
       ℃
    </span>
    <button
      onClick={[Function]}
    >
      X
    </button>
  </div>
  <List
    city="Ufa"
    image="http://openweathermap.org/img/wn/04n@2x.png"
    isLoading={null}
    temp="-6"
  />
</div>
`;
